apps/dashr-studybrowser/app.R

```diff
@@ -0,0 +1,286 @@
+
+app$layout(
+  
+  htmlDiv(className='', children=l.(
+    
+    #Title
+    htmlDiv(className='study-browser-banner', children="Animal Study Browser")
+    ,
+    htmlDiv(className='container', children= l.(
+      
+      htmlDiv(className='row', children= l.(
+        
+        htmlDiv(className='six columns', children=l.(
+          
+          htmlEm('Test Article:')
+          ,
+          dccDropdown(id='study-dropdown', options=l.(), value='')
+        
+        ))
+        ,
+        htmlDiv(className='six columns', children = l.())
+        
+        
+      ))
+      ,
+      htmlDiv(className='row', children=l.(
+        
+        htmlDiv(className='ten columns', children=l.(
+          
+          dccGraph(id='plot', figure=default_plot)
+          
+        )
+        
+       
+        )
+        ,
+        htmlDiv(className='two columns', children=l.(
+          
+          dccRadioItems(id='chart-type', options=l.(l.('label'='Box Plot', 'value'='box'), l.('label'='Violin Plot', 'value'='violin')),  value='violin')
+          
+        ))
+        
+      ))
+      ,
+      
+      htmlDiv(className='row', style=l.(), children=l.(
+        
+        htmlDiv(className='twelve columns', children=l.(
+          
+          dccUpload(
+            
+            id='upload-data'
+            ,
+            children= htmlDiv(l.("Drag and Drop or ", htmlA("Select Files")))
+            ,
+            style=l.(
+              
+              'width'='100%'
+              ,
+              'height'='60px'
+              ,
+              'lineHeight'='60px'
+              ,
+              'borderWidth'='1px'
+              ,
+              'borderStyle'='dashed'
+              ,
+              'borderRadius'='5px'
+              ,
+              'textAlign'='center'
+              ,
+              'margin'='10px'
+            )
+            ,
+            filename=''
+            
+          
+          )
+          ,
+          htmlP("", id='file_checker')
+          ,
+          htmlP(" 0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 '!' 1 ", style=l.(display='block', width='100%'))
+          
+        ))
+        
+      ))
+    
+    ))
+    
+  ))
+  
+)
+
+app$callback(
+  output=l.(id='file_checker', property='children')
+  ,
+  params=l.(input(id='upload-data', property='filename'))
+  ,
+  function(filename){
+    if(is.na(filename)|| is.null(filename)||(filename=='')){
+      return("No file ")
+    } else{
+      return(glue('Data {filename} has been successfully uploaded'))
+    }
+  }
+  
+  
+)
+#Decoding and parsing the file uploaded:
+#helper function
+parse_contents <- function(contents, filename){
+  contents_parsed <- strsplit(contents, ',')
+  decoded <- jsonlite::base64_dec(contents_parsed[[1]][2]) %>% rawToChar(.)
+  if(str_detect(filename, '.csv')){
+    DT <- fread(decoded)
+    
+  } else if(str_detect(filename, '.xls')){
+    DT <- read_xls(decoded)
```

Comment:
Do we need Excel spreadsheet support? I don't think this was in the original Python version.

Reply:
This is copied from some of my other apps;